test/e2e/dns_configmap.go

```diff
@@ -0,0 +1,314 @@
+			// Check local mapping. Checking a remote mapping requires
+			// creating an arbitrary DNS record which is not possible at the
+			// moment.
+			By(fmt.Sprintf("Validating federation record %v", label))
+			predicate := func(actual []string) bool {
+				for _, v := range actual {
+					if v == localDNS {
+						return true
+					}
+				}
+				return false
+			}
+			t.checkDNSRecord(federationDNS, predicate, timeout)
+		}
+	}
+}
+
+func (t *dnsConfigMapTest) checkDNSRecord(name string, predicate func([]string) bool, timeout time.Duration) {
+	startTime := time.Now()
+	ok := false
+	var actual []string
+
+	for time.Since(startTime) < timeout {
+		time.Sleep(500 * time.Millisecond)
+		actual = t.runDig(name)
+		if predicate(actual) {
+			ok = true
+			break
+		}
+	}
+
+	if !ok {
+		framework.Logf("dig result did not match: %#v, waited %v", actual, timeout)
+	}
+
+	Expect(ok).Should(BeTrue())
+}
+
+// runDig querying for `dnsName`. Returns a list of responses.
+func (t *dnsConfigMapTest) runDig(dnsName string) []string {
+	cmd := []string{
+		"/usr/bin/dig",
+		"+short",
+		"@" + t.dnsPod.Status.PodIP,
+		"-p", "10053", dnsName,
+	}
+	stdout, stderr, err := t.f.ExecWithOptions(framework.ExecOptions{
+		Command:       cmd,
+		Namespace:     t.f.Namespace.Name,
+		PodName:       t.utilPod.Name,
+		ContainerName: "util",
+		CaptureStdout: true,
+		CaptureStderr: true,
+	})
+
+	Expect(err).NotTo(HaveOccurred())
+
+	framework.Logf("Running dig: %v, stdout: %q, stderr: %q",
+		cmd, stdout, stderr)
+
+	if stdout == "" {
+		return []string{}
+	} else {
+		return strings.Split(stdout, "\n")
+	}
+}
+
+func (t *dnsConfigMapTest) setConfigMap(cm *api.ConfigMap, isValid bool) {
+	if isValid {
+		t.cm = cm
+	}
+	t.isValid = isValid
+
+	cm.ObjectMeta.Namespace = t.ns
+	cm.ObjectMeta.Name = t.name
+
+	options := api.ListOptions{
+		FieldSelector: fields.Set{
+			"metadata.namespace": t.ns,
+			"metadata.name":      t.name,
+		}.AsSelector(),
+	}
+	cmList, err := t.c.Core().ConfigMaps(t.ns).List(options)
+	Expect(err).NotTo(HaveOccurred())
+
+	if len(cmList.Items) == 0 {
+		By(fmt.Sprintf("Creating the ConfigMap (%s:%s) %+v", t.ns, t.name, *cm))
+		_, err := t.c.Core().ConfigMaps(t.ns).Create(cm)
+		Expect(err).NotTo(HaveOccurred())
+	} else {
+		By(fmt.Sprintf("Updating the ConfigMap (%s:%s) to %+v", t.ns, t.name, *cm))
+		_, err := t.c.Core().ConfigMaps(t.ns).Update(cm)
+		Expect(err).NotTo(HaveOccurred())
+	}
+}
+
+func (t *dnsConfigMapTest) deleteConfigMap() {
+	By(fmt.Sprintf("Deleting the ConfigMap (%s:%s)", t.ns, t.name))
+
+	t.cm = nil
+	t.isValid = false
+
+	err := t.c.Core().ConfigMaps(t.ns).Delete(t.name, nil)
+	Expect(err).NotTo(HaveOccurred())
+}
+
+func (t *dnsConfigMapTest) createUtilPod() {
+	t.utilPod = &api.Pod{
+		TypeMeta: unversioned.TypeMeta{
+			Kind: "Pod",
+		},
+		ObjectMeta: api.ObjectMeta{
+			Namespace: t.f.Namespace.Name,
+			Name:      "e2e-dns-configmap-" + string(uuid.NewUUID()),
```

Comment:
You can set GenerateName. Example: https://github.com/kubernetes/kubernetes/blob/33ebe1f18b9cf5909931376f656e19e80ac9ac1c/test/e2e/service_latency.go#L320